### PR TITLE
Halt on calling brk() in a pthreads build, as it is not threadsafe yet

### PR DESCRIPTION
**Pull request by kripken**
See #10006

### CODE REVIEW

File: system/lib/sbrk.c [hunk 1]
```diff
@@ -99,6 +101,11 @@ void *sbrk(intptr_t increment) {
 }
 
 int brk(intptr_t ptr) {
+#if __EMSCRIPTEN_PTHREADS__
```
Comment (dschuff) [inline system/lib/sbrk.c]: I guess we could even make this work only on the main thread (but not background threads) if we wanted.

**Comment by VirtualTim** [comment]
I don't really like this change.
If brk is called from malloc this will abort, even though it's safe to do so.
Also my understanding of the impact of the racy-ness is that it's quite low. It it is hit the impact is that the heap is increased by more than it should have been, but nothing will really be broken.

What do you think about putting this behind `ASSERTIONS`? Or am I underestimating the impact?

File: system/lib/sbrk.c [hunk 2]
```diff
@@ -11,7 +11,9 @@
 #endif
 #include <limits.h>
 #include <stddef.h>
+#include <stdlib.h>
 #include <stdint.h>
+#include <stdio.h>
```
Comment (VirtualTim) [inline system/lib/sbrk.c]: These newly added headers should probably also be behind an `#if __EMSCRIPTEN_PTHREADS__`.

**Comment by dschuff** [comment]
`brk` is not actually called by malloc though, it's `sbrk`. IIUC any calls of `brk` would be in user code, which means that the user would have particular expectations (i.e. they'd expect it to work, which it might not).

**Comment by kripken** [comment]
Correct, our default malloc doesn't call brk, only sbrk.

@VirtualTim This patch just avoids users having silent memory races or other issues, if they ship their own allocation code that does use brk, until we fix this properly.

**Comment by kripken** [comment]
It is true that if the user ships a malloc that is threadsafe (like dlmalloc), and that does use brk, then this PR would make it fail. But we can't tell the custom allocator is safe.

**Comment by VirtualTim** [comment]
OK, I guess I misunderstood the impact. If this only affects brk called from user code then this change is good.

**Pull request merged**